ignite/engine/engine.py

```diff
@@ -332,23 +278,7 @@ def has_event_handler(self, handler: Callable, event_name: Optional[Any] = None)
             event_name: The event the handler attached to. Set this
                 to ``None`` to search all events.
         """
-        if event_name is not None:
-            if event_name not in self._event_handlers:
-                return False
-            events = [event_name]  # type: Union[List[Any], Dict[Any, List]]
-        else:
-            events = self._event_handlers
-        for e in events:
-            for h, _, _ in self._event_handlers[e]:
-                if self._compare_handlers(handler, h):
-                    return True
-        return False
-
-    @staticmethod
-    def _compare_handlers(user_handler: Callable, registered_handler: Callable) -> bool:
-        if hasattr(registered_handler, "_parent"):
-            registered_handler = registered_handler._parent()  # type: ignore[attr-defined]
-        return registered_handler == user_handler
+        return super(Engine, self).has_event_handler(handler, event_name=event_name)
```

Comment:
Same here, useless ?

Reply:
Yes, all the functions left in the Engine is left for the keeping the docs.